Classes/Emogrifier.php

```diff
@@ -296,12 +260,59 @@ public function emogrify()
             $this->removeInvisibleNodes($xpath);
         }
 
-        $this->copyCssWithMediaToStyleNode($cssParts, $xmlDocument);
+        $this->copyCssWithMediaToStyleNode($cssParts['media'], $xmlDocument);
 
         return $xmlDocument->saveHTML();
     }
 
     /**
+     * Parse a list of selectors from a string of css.
+     *
+     * @param string[] $css a string of raw CSS code
+     *
+     * @return array
+     */
+    private function parseSelectors($css)
+    {
+        $cssKey = md5($css);
+        if (!isset($this->caches[self::CACHE_KEY_CSS][$cssKey])) {
+            // process the CSS file for selectors and definitions
+            preg_match_all('/(?:^|[\\s^{}]*)([^{]+){([^}]*)}/mis', $css, $matches, PREG_SET_ORDER);
+
+            $allSelectors = array();
+            foreach ($matches as $key => $selectorString) {
+                // if there is a blank definition, skip
+                if (!strlen(trim($selectorString[2]))) {
```

Comment:
Please use !== '' instead of strlen as this makes the intent of the code more clear.